### ISSUE TITLE
Farsi is not in the input table list in braille settings?

### ISSUE THREAD

**Issue by parhamdoustdar** [nvaccess/nvda#6231]
Even though liblouis authors say that the braille table definitions are used both for input and output, NVDA does not have Farsi as an option in the braille input table. Is it possible to add Farsi to input options?

**Comment by josephsl** [comment]
Hi, are you sure this table uses one-to-one mapping between chars and cells, like that of United States computer braile code? If not, this cannot be included.

**Comment by mohammadrezarashad** [comment]
Dear Joseph, can you explain me one-to-one mapping a bit more? Do you mean we should one-cell, and just one-cell  Braille equivalent for each character? I corrected the existing Farsi braille table and want to include it as a Braille input table also. Please help me so I can revise the table to meet acceptable standard for Braille input tables in NVDA.

**Comment by jcsteh** [comment]
NVDA currently only supports braille input for computer braille tables. That is, each single character must map to a single braille cell pattern; multiple cells aren't possible, nor are multiple characters mapping to the same cell pattern. This simply isn't possible for some languages, but regardless, this is currently what is required for NVDA. Note that support for uncontracted and contracted braille is being worked on; see #2439.

**Comment by mohammadrezarashad** [comment]
Dear Jamie,
Many many thanks for your answer. I'll try to create a separate table
for Farsi according to this rule.

۱۳۹۵-۰۵-۲۱ ۶:۰۶ +۰۴:۳۰ گرینویچ, James Teh <notifications@github.com>:

**Comment by mohammadrezarashad** [comment]
Dear friends, I created my Farsi Braille table. I will create my Braille input table for Farsi soon. How can I test these tables with NVDA?

**Comment by mohammadrezarashad** [comment]
Dear friends, I created an updated Braille table and also a computer Braille table for Farsi, and sent them to Liblouis mailing list. I'm waiting for their reply.

**Comment by jcsteh** [comment]
Now that contracted braille input (#6449) is in master, NVDA can support input of computer braille, uncontracted braille and contracted braille. Computer braille and uncontracted braille are handled the same way. However, NVDA needs to distinguish contracted braille tables.

If you want new tables added, please provide the following information for each table: the table file name, a description to be shown in the braille tables list and whether it is a contracted braille table. Contracted braille means that one braille cell might produce multiple characters.

Also, there is a table labelled Farsi grade 1 (ar-fa.utb). Should this table be kept? Should its description be changed? I don't quite understand the distinction between Persian and Farsi braille.
